processor/span/factory.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package span
+
+import (
+	"errors"
+
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-service/config/configerror"
+	"github.com/open-telemetry/opentelemetry-service/config/configmodels"
+	"github.com/open-telemetry/opentelemetry-service/consumer"
+	"github.com/open-telemetry/opentelemetry-service/processor"
+)
+
+const (
+	// typeStr is the value of "type" Span Rename in the configuration.
+	typeStr = "span"
+)
+
+// errMissingRequiredField is returned when a required field in the config
```

Comment:
Seems like a good candidate to be moved to `oterr` package, but it needs to be something that takes the respective tags as parameters to generate proper message. Leave as it is in this PR: there is already issue https://github.com/open-telemetry/opentelemetry-service/issues/215 to track this

Reply:
I'll add a todo here to reference the issue.